### PR TITLE
feat: Add dimension prop to Combo

### PR DESCRIPTION
**Pull request by pratyushbanerjee**
## Description

Adds `dimension` prop to Combo which is passed on to `Bar` and `Line`

<div>
  <table>
  <tr>
  <th>Before</th>
  <th>After</th>
  </tr>
  <tr>
  <td>

  ```tsx
  <Combo>
    <Bar
      dimension="datetime"
      metric="people"
      metricAxis="people" 
    />
    <Line 
      dimension="datetime"
      metric="adoptionRate" 
      metricAxis="adoption"
    />
  </Combo>
  ```

  </td>
  <td>

  ```tsx
  <Combo dimension="datetime">
    <Bar 
      metric="people"
      metricAxis="people" 
    />
    <Line 
      metric="adoptionRate" 
      metricAxis="adoption"
    />
  </Combo>
  ```

  </td>
  </tr>
  </table>
</div>

## Related Issue

https://github.com/adobe/react-spectrum-charts/issues/364

## Motivation and Context

Provides a simpler API to set the common dimension of the Combo chart

## How Has This Been Tested?

- Passed existing tests
- Updated tests to cover changes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by pratyushbanerjee** [comment]
Can we skip coverage requirement for pseudo component?

**Comment by marshallpete** [comment]
You could do something like we have in `Bar.test.tsx`. You will see that the first test just renders the pseudo component. That is enough to count it as covered. If for some reason it isn't a react component, then that test would fail to render.

I would create a file called `Combo.test.tsx` in `src/stories/components/Combo/` and add the test to that. Seems like a silly test but it does at least ensure that the component is a react component.

**Comment by pratyushbanerjee** [comment]
Thanks! Added the tests

File: src/specBuilder/combo/comboSpecBuilder.ts [hunk 1]
```diff
@@ -20,14 +20,17 @@ import { Spec } from 'vega';
...
+		{ children = [], colorScheme = DEFAULT_COLOR_SCHEME, index = 0, name, dimension = DEFAULT_TIME_DIMENSION }
+	) => {
 		const buildOrder = new Map();
 		buildOrder.set(Bar, 0);
 		buildOrder.set(Line, 0);
 
 		let { barCount, lineCount } = initializeComponentCounts();
 		const sanitizedChildren = sanitizeRscChartChildren(children);
-		const comboName = toCamelCase(name || `combo${index}`);
+		const comboName = toCamelCase(name ?? `combo${index}`);
```
Comment (marshallpete) [inline src/specBuilder/combo/comboSpecBuilder.ts:33]:
```suggestion
		const comboName = toCamelCase(name || `combo${index}`);
```

Comment (marshallpete) [reply src/specBuilder/combo/comboSpecBuilder.ts:33]: I actually prefer this as an OR (`||`) here. If the name is an empty string, we want that to be false and use the default name instead.
Nullish coellesence (spelling?) will treat an empty string as true and use the empty string as the name.

I know this fails the code smells but I just mark them as accepted in sonarCloud. I'll see if I can figure out how to ignore this check in sonarCloud.

**Review by marshallpete — APPROVED** [review]

**Pull request merged**